File: miraculous-frontend/miraculous-ui/src/dto/full-track-info-dto/full-track-info-dto.ts
export class FullTrackInfoDto {

  id: number;
  title: string;
  artistId: number;
  artistLogin: string;
  albumId: number;
  albumTitle: string;
  dateOfRelease: string;
  duration: number;
  genres: string[];
  description: string;
  isFree: boolean;
  price: number;
  numberOfLikes: number;
  audioUrl: string;
  coverPictureUrlName: string;
  size: number;
  inCartOfCurrentVisitor: boolean;
  likedByCurrentVisitor: boolean;


  constructor(id?: number, title?: string, artistId?: number, artistLogin?: string, albumId?: number,
              albumTitle?: string, dateOfRelease?: string, duration?: number, genres?: string[],
              description?: string, isFree?: boolean, price?: number, numberOfLikes?: number, audioUrl?: string,
              coverPictureUrl?: string, size?: number, isInCurrentVisitorCart?: boolean, likedByCurrentVisitor?: boolean) {
    this.id = id;
    this.title = title;
    this.artistId = artistId;
    this.artistLogin = artistLogin;
    this.albumId = albumId;
    this.albumTitle = albumTitle;
    this.dateOfRelease = dateOfRelease;
    this.duration = duration;
    this.genres = genres;
    this.description = description;
    this.isFree = isFree;
    this.price = price;
    this.numberOfLikes = numberOfLikes;
    this.audioUrl = audioUrl;
    this.coverPictureUrlName = coverPictureUrl;
    this.size = size;
    this.inCartOfCurrentVisitor = isInCurrentVisitorCart;
    this.likedByCurrentVisitor = likedByCurrentVisitor;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/FullTrackInfoDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FullTrackInfoDto {

    private Long id;
    private String title;
    private Long artistId;
    private String artistLogin;
    private Long albumId;
    private String albumTitle;
    private String dateOfRelease;
    private Double duration;
    private List<String> genres;
    private String description;
    private Boolean isFree;
    private Double price;
    private Long numberOfLikes;
    private String audioUrl;
    private String coverPictureUrlName;
    private Double size;
    private boolean isLikedByCurrentVisitor;
    private boolean isInCurrentVisitorCart;
}
